### ISSUE TITLE
Get radiometric calibration array

### ISSUE THREAD

**Issue by ebuchlin** [solo-spice/sospice#50]
Get radiometric calibration array as a function of wavelength, for a given spectral window, equivalent to the [attached IDL code](https://github.com/solo-spice/sospice/files/14779051/get_radcal_array.txt) from T. Fredvik:

```idl
IDL> radcal_array = get_radcal_array(l2_file, win_no, scalar_value=scalar_value)
IDL> help,radcal_array
RADCAL_ARRAY    DOUBLE    = Array[1, 1, 80, 1]
IDL> print,scalar_value
       1647.6411
IDL> print,mean(radcal_array)
       1647.6411
```

**Comment by adolliou** [comment]
I propose to set the three IDL functions rewritten into python in a radcal.py file in util.

**Comment by ebuchlin** [comment]
We could put the new `radcal.py` in `calibrate/` rather than `util/`.

`get_radcal_column_no()` and `get_radcal_column_no()` would be private functions.

I would work as much as possible on a `HDUList` instead of a filename in order to avoid opening the same file several times (`get_radcal_array()` could still offer the choice to take a filename or `HDUList` as first argument). The `exten_no` parameter could be an index or an HDU name (or several indices or HDU names?).

**Comment by adolliou** [comment]
Yes I will go in this direction. The HDULISt should only be opened once. 

I suggest a method get_radcal_array_from_hdu which call use as an input a HDU. Then I define another public method get_radcal_array_ taking different input types, that calls get_radcal_array_from_hdu : 

the input types  could be : 
- int, str (extname) + path to fits file : returns a radcal array of the corresponding window
- list  (length N) of str or int + path to fits file : return a list (length N) of radcal arrays of the corresponding windows
- HDU that simply calls get_radcal_array_from_hdu ? 
+ Checks that the inputs are consistent.

Or maybe something simpler ?

**Comment by ebuchlin** [comment]
An HDU is not enough, as you need both a spectral window HDU and the `VARIABLE_KEYWORDS` HDU. But I agree you could have a function taking both these HDUs as an argument, and a wrapper function doing some magic with different types of user-provided arguments.

**Comment by adolliou** [comment]
Hi, I progress, but I just have one question.

The radcal array in the binary table is the same size as the wavelength array in the window WCS (as it should be). In that case, would it be better not to add a wavelength array as input of the function, but simply return the radcal_array for the given window (eventually with the associated wavelength array computed with WCS, which could be returned with an option). 

If I understand correctly, having as input a personalized wavelength array would require to do an interpolation on the radcal_array, which does not make much sense anyway.

Bests,

Antoine

**Comment by ebuchlin** [comment]
Yes, no need for a wavelength array (I don't think that the IDL code has one?)

**Comment by adolliou** [comment]
See PR https://github.com/solo-spice/sospice/pull/73
